cmd/mailbox-controller/main.go

```diff
@@ -95,54 +91,58 @@ func main() {
 		}
 	}()
 
-	// create edgeSharedInformerFactory
-	espwRestConfig, err := espwClientOpts.ToRESTConfig()
+	// create space-aware client
+	spaceManagementConfig, err := spaceMgtOpts.ToRESTConfig()
 	if err != nil {
-		logger.Error(err, "failed to create config from flags")
+		logger.Error(err, "Failed to create space management config from flags")
 		os.Exit(3)
 	}
-
-	edgeClientset, err := edgeclientset.NewForConfig(espwRestConfig)
+	spaceclient, err := spaceclient.NewMultiSpace(ctx, spaceManagementConfig)
 	if err != nil {
-		logger.Error(err, "failed to create clientset for view of edge exports")
-		os.Exit(6)
+		logger.Error(err, "Failed to create space-aware client")
+		os.Exit(4)
 	}
+	spaceProviderNs := spacemanager.ProviderNS(spaceProvider)
 
-	edgeSharedInformerFactory := edgeinformers.NewSharedScopedInformerFactoryWithOptions(edgeClientset, resyncPeriod)
-	syncTargetClusterPreInformer := edgeSharedInformerFactory.Edge().V2alpha1().SyncTargets()
+	coreRestConfig, err := spaceclient.ConfigForSpace(coreSpace, spaceProviderNs)
+	if err != nil {
+		logger.Error(err, "Failed to fetch space config", "spacename", coreSpace)
+		os.Exit(5)
+	}
 
-	rootRestConfig, err := rootClientOpts.ToRESTConfig()
+	edgeClientset, err := edgeclientset.NewForConfig(coreRestConfig)
 	if err != nil {
-		logger.Error(err, "failed to make root config")
-		os.Exit(8)
+		logger.Error(err, "Failed to create edge clientset for KubeStellar Core Space")
+		os.Exit(6)
 	}
-	rootRestConfig.UserAgent = "mailbox-controller"
```

Comment:
Setting the UserAgent is good. It helps debugging.

Reply:
It would be better if that were done before the usage in `edgeClientset, err := edgeclientset.NewForConfig(kcsRestConfig)`.

Reply:
This is a pervasive issue. Probably should be part of the msclient's functionality.